File: sc-moyuan-backend/src/main/java/com/moyuan/entity/PoetDraft.java
package com.moyuan.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("poet_draft")
public class PoetDraft {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long poetId;

    private String section;

    private String content;

    private String originalContent;

    private Long editorId;

    private Integer status;

    private Long reviewerId;

    private String reviewComment;

    private LocalDateTime reviewTime;

    private String ip;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: frontend/src/api/modules/poet-draft.ts
import request from '@/utils/request'

export interface PoetDraft {
  id?: number
  poetId: number
  section: string
  content: string
  originalContent?: string
  editorId?: number
  status?: number
  reviewerId?: number
  reviewComment?: string
  reviewTime?: string
  ip?: string
  createTime?: string
  updateTime?: string
}

export function savePoetDraft(data: { poetId: number; section: string; content: string }) {
  return request.post('/poet-drafts', data)
}

export function getAdminPoetDrafts(params: { pageNum?: number; pageSize?: number; poetId?: number; status?: number }) {
  return request.get<{ records: PoetDraft[]; total: number }>('/admin/poet-drafts', { params })
}

export function getAdminPoetDraftById(id: number) {
  return request.get<PoetDraft>(`/admin/poet-drafts/${id}`)
}

export function reviewPoetDraft(id: number, status: number, reviewComment?: string) {
  return request.put(`/admin/poet-drafts/${id}/review`, null, {
    params: { status, reviewComment }
  })
}
